### PR TITLE
Adding documentation for remote index use in AD

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description
In 2.17 we have added the capability to select a remote index as a data source for creating an anomaly detector
### Issues Resolved

Closes https://github.com/opensearch-project/documentation-website/issues/8169

### Version
2.17
### Frontend features
new addition is just the cluster drop down

Screenshot 2024-09-06 at 8 02 55 AM

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by jackiehanyang — APPROVED** [review]

File: _observing-your-data/ad/index.md [hunk 1]
```diff
@@ -29,9 +29,14 @@ A detector is an individual anomaly detection task. You can define multiple dete
 1. Add in the detector details.
    - Enter a name and brief description. Make sure the name is unique and descriptive enough to help you to identify the purpose of the detector.
 1. Specify the data source.   
-   - For **Data source**, choose the index you want to use as the data source. You can optionally use index patterns to choose multiple indexes.
+   - For **Data source**, choose the index/indices you want to use as the data source. You can also optionally use an alias or an index patterns to choose multiple indexes.
```
Comment (vagimeli) [inline _observing-your-data/ad/index.md:32]:
```suggestion
   - For **Data source**, choose the index or indexes you want to use as the data source. You can also optionally use an alias or an index patterns to choose multiple indexes. 
```

File: _observing-your-data/ad/index.md [hunk 2]
```diff
@@ -29,9 +29,14 @@ A detector is an individual anomaly detection task. You can define multiple dete
 1. Add in the detector details.
    - Enter a name and brief description. Make sure the name is unique and descriptive enough to help you to identify the purpose of the detector.
 1. Specify the data source.   
-   - For **Data source**, choose the index you want to use as the data source. You can optionally use index patterns to choose multiple indexes.
+   - For **Data source**, choose the index or indexes you want to use as the data source. You can also optionally use an alias or an index patterns to choose multiple indexes. 
+   - Detectors also support remote indices, you can utilize the same `cluster-name:index-name` pattern used by [cross-cluster search](https://opensearch.org/docs/latest/search-plugins/cross-cluster-search/) or by utilizing OpenSearch Dashboards 2.17 or later to choose the clusters and indices you want to use.
```
Comment (vagimeli) [inline _observing-your-data/ad/index.md:33]:
```suggestion
   - Detectors can use remote indexes. You can access them using the `cluster-name:index-name` pattern, as used in [cross-cluster search](https://opensearch.org/docs/latest/search-plugins/cross-cluster-search/). Alternatively, you can select the clusters and indexes you want to use through OpenSearch Dashboards 2.17 or later.
```

File: _observing-your-data/ad/index.md [hunk 3]
```diff
@@ -29,9 +29,14 @@ A detector is an individual anomaly detection task. You can define multiple dete
...
+   - Detectors can use remote indexes. You can access them using the `cluster-name:index-name` pattern, as used in [cross-cluster search](https://opensearch.org/docs/latest/search-plugins/cross-cluster-search/). Alternatively, you can select the clusters and indexes you want to use through OpenSearch Dashboards 2.17 or later.
    - (Optional) For **Data filter**, filter the index you chose as the data source. From the **Data filter** menu, choose **Add data filter**, and then design your filter query by selecting **Field**, **Operator**, and **Value**, or choose **Use query DSL** and add your own JSON filter query. Only [Boolean queries]({{site.url}}{{site.baseurl}}/query-dsl/compound/bool/) are supported for query domain-specific language (DSL).
 
+
+The following [permissions]({{site.url}}{{site.baseurl}}/security/access-control/permissions/) are required in order to create a cross-cluster detector through the dashboards UI: `indices:data/read/field_caps`, `indices:admin/resolve/index`, and `cluster:monitor/remote/info`.
```
Comment (vagimeli) [inline _observing-your-data/ad/index.md:37]:
```suggestion
To create a cross-cluster detector through OpenSearch Dashboards, the following [permissions]({{site.url}}{{site.baseurl}}/security/access-control/permissions/) are required: `indices:data/read/field_caps`, `indices:admin/resolve/index`, and `cluster:monitor/remote/info`.
```

File: _observing-your-data/ad/index.md [hunk 4]
```diff
@@ -29,9 +29,14 @@ A detector is an individual anomaly detection task. You can define multiple dete
 1. Add in the detector details.
    - Enter a name and brief description. Make sure the name is unique and descriptive enough to help you to identify the purpose of the detector.
 1. Specify the data source.   
-   - For **Data source**, choose the index you want to use as the data source. You can optionally use index patterns to choose multiple indexes.
+   - For **Data source**, choose the index or indexes you want to use as the data source. You can also optionally use an alias or an index patterns to choose multiple indexes.
```
Comment (vagimeli) [inline _observing-your-data/ad/index.md:32]:
```suggestion
   - For **Data source**, choose the index or indexes you want to use as the data source. Alternatively, you can use an alias or index patterns to choose multiple indexes. 
```

**Review by vagimeli — APPROVED** [review]
Doc review complete. Edits made for clarity and compliance with OpenSearch style.

File: _observing-your-data/ad/security.md [hunk 5]
```diff
@@ -31,6 +36,36 @@ When a trigger generates an alert, the detector and monitor configurations, the
 
 To reduce the chances of unintended users viewing metadata that could describe an index, we recommend that administrators enable role-based access control and keep these kinds of design elements in mind when assigning permissions to the intended group of users. See [Limit access by backend role](#advanced-limit-access-by-backend-role) for details.
 
+
+### Selecting remote indices with fine-grained access control
+
+In order to specify a remote index as the data source for a detector you must follow similar instruction as mentioned in the cross-cluster search documentation. By utilizing a role that exists in the both the remote and local cluster.The remote cluster will need to have the mapping between the chosen role and the same user name as in the local cluster.
```
[the remark this answers is not in the document]
Comment (amitgalitz) [reply _observing-your-data/ad/security.md]: added

File: _observing-your-data/ad/security.md [hunk 6]
```diff
@@ -31,6 +36,36 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
+On the remote cluster, create the same user, and map anomaly_full_access to that role:
+
+```
+curl -XPUT -k -u 'admin:<custom-admin-password>' 'https://localhost:9250/_plugins/_security/api/internalusers/anomalyuser' -H 'Content-Type: application/json' -d '{"password":"password"}'
+curl -XPUT -k -u 'admin:<custom-admin-password>' -H 'Content-Type: application/json' 'https://localhost:9250/_plugins/_security/api/rolesmapping/anomaly_full_access' -d '{"users" : ["anomalyuser"]}'
+```
+
+### Custom result index
+
+In order to use a custom result index users will need to have additional permissions that aren't part of the default roles that OpenSearch Security plugin offers. Users will have to add additional permissions as mentioned in the custom result index set up [step]({{site.url}}{{site.baseurl}}/observing-your-data/ad/index/#example-filter-using-query-dsl:~:text=Specify%20custom%20results%20index.).
```
Comment (kaituo) [inline _observing-your-data/ad/security.md:67]: Have you tested using Jekyll build whether these links work?  I tried to use observing-your-data in https://github.com/opensearch-project/documentation-website/pull/8202 but it does not work.

Comment (amitgalitz) [reply _observing-your-data/ad/security.md:67]: This link should work, I checked with jekyll

**Comment by amitgalitz** [comment]
Jekyll is failing but links works locally fine for me, I can just delete links if its a blocker

**Comment by kaituo** [comment]
> Jekyll is failing but links works locally fine for me, I can just delete links if its a blocker

Can you try {{site.url}}{{site.baseurl}}/monitoring-plugins/ad/... or change back to original link?

**Review by kaituo — APPROVED** [review]

File: _observing-your-data/ad/security.md [hunk 7]
```diff
@@ -31,6 +36,36 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
 To reduce the chances of unintended users viewing metadata that could describe an index, we recommend that administrators enable role-based access control and keep these kinds of design elements in mind when assigning permissions to the intended group of users. See [Limit access by backend role](#advanced-limit-access-by-backend-role) for details.
 
+
+### Selecting remote indexes with fine-grained access control
+
+In order to specify a remote index as the data source for a detector you must follow similar instruction as mentioned in the cross-cluster search [documentation]({{site.url}}{{site.baseurl}}/search-plugins/cross-cluster-search/#authentication-flow). By utilizing a role that exists in the both the remote and local cluster.The remote cluster will need to have the mapping between the chosen role and the same user name as in the local cluster. 
+
+For example:
+
+Create a new user on the local cluster from which you’ll create detectors from:
```
Comment (natebower) [inline _observing-your-data/ad/security.md:46]: Please revise.

File: _observing-your-data/ad/security.md [hunk 8]
```diff
@@ -31,6 +36,36 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
+
+```
+curl -XPUT -k -u 'admin:<custom-admin-password>' 'https://localhost:9200/_plugins/_security/api/internalusers/anomalyuser' -H 'Content-Type: application/json' -d '{"password":"password"}'
+```
+
+Map the new user to the anomaly_full_access role:
```
Comment (natebower) [inline _observing-your-data/ad/security.md:52]:
```suggestion
Map the new user to the `anomaly_full_access` role:
```

File: _observing-your-data/ad/security.md [hunk 9]
```diff
@@ -31,6 +36,36 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
+
+```
+curl -XPUT -k -u 'admin:<custom-admin-password>' -H 'Content-Type: application/json' 'https://localhost:9200/_plugins/_security/api/rolesmapping/anomaly_full_access' -d '{"users" : ["anomalyuser"]}'
+```
+
+On the remote cluster, create the same user, and map anomaly_full_access to that role:
```
Comment (natebower) [inline _observing-your-data/ad/security.md:58]:
```suggestion
On the remote cluster, create the same user and map `anomaly_full_access` to that role:
```

File: _observing-your-data/ad/security.md [hunk 10]
```diff
@@ -31,6 +36,36 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
+
+```
+curl -XPUT -k -u 'admin:<custom-admin-password>' 'https://localhost:9250/_plugins/_security/api/internalusers/anomalyuser' -H 'Content-Type: application/json' -d '{"password":"password"}'
+curl -XPUT -k -u 'admin:<custom-admin-password>' -H 'Content-Type: application/json' 'https://localhost:9250/_plugins/_security/api/rolesmapping/anomaly_full_access' -d '{"users" : ["anomalyuser"]}'
+```
+
+### Custom result index
```
Comment (natebower) [inline _observing-your-data/ad/security.md:65]:
```suggestion
### Custom results index
```

File: _observing-your-data/ad/security.md [hunk 11]
```diff
@@ -31,6 +36,36 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
+
+In order to use a custom result index users will need to have additional permissions that aren't part of the default roles that OpenSearch Security plugin offers. Users will have to add additional permissions as mentioned in the custom result index set up step in [define detector]({{site.url}}{{site.baseurl}}/observing-your-data/ad/index/#step-1-define-a-detector).
```
Comment (natebower) [inline _observing-your-data/ad/security.md:67]:
```suggestion
In order to use a custom results index, users need additional permissions that aren't included in the default roles provided by the OpenSearch Security plugin. Users need to add additional permissions, as mentioned in the custom result index setup step in [define detector]({{site.url}}{{site.baseurl}}/observing-your-data/ad/index/#step-1-define-a-detector).
```

File: _observing-your-data/ad/security.md [hunk 12]
```diff
@@ -31,6 +36,36 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
+
```
Comment (natebower) [inline _observing-your-data/ad/security.md:68]: Line above: Last sentence needs some revision.

**Review by natebower — CHANGES_REQUESTED** [review]
@amitgalitz @vagimeli Please see my comments and changes and tag me when complete for a reread and approval. Thanks!

File: _observing-your-data/ad/security.md [hunk 13]
```diff
@@ -23,6 +23,11 @@ As an admin user, you can use the Security plugin to assign specific permissions
 
 The Security plugin has two built-in roles that cover most anomaly detection use cases: `anomaly_full_access` and `anomaly_read_access`. For descriptions of each, see [Predefined roles]({{site.url}}{{site.baseurl}}/security/access-control/users-roles#predefined-roles).
 
+If you are using OpenSearch Dashboards to create your anomaly detectors you might notice issues creating a detector even with `anomaly_full_access`, this has been mitigated in OpenSearch 2.17. In order to mitigate this issue in prior versions, you should add these additional permissions:
+
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _observing-your-data/ad/security.md:27]: @amitgalitz Please see the editor's comment above. We should give the users context about what issues they may experience.

File: _observing-your-data/ad/security.md [hunk 14]
```diff
@@ -31,6 +36,42 @@ When a trigger generates an alert, the detector and monitor configurations, the
 
 To reduce the chances of unintended users viewing metadata that could describe an index, we recommend that administrators enable role-based access control and keep these kinds of design elements in mind when assigning permissions to the intended group of users. See [Limit access by backend role](#advanced-limit-access-by-backend-role) for details.
 
+### Selecting remote indexes with fine-grained access control
+
+To use a remote index as a data source for a detector, see [Authentication flow]({{site.url}}{{site.baseurl}}/search-plugins/cross-cluster-search/#authentication-flow) in [Cross-cluster search]({{site.url}}{{site.baseurl}}/search-plugins/cross-cluster-search/). You must use a role that exists in both the remote and local clusters. The remote cluster must map the chosen role to the same username as in the local cluster.
```
Comment (vagimeli) [inline _observing-your-data/ad/security.md:41]:
```suggestion
To use a remote index as a data source for a detector, see the setup steps in [Authentication flow]({{site.url}}{{site.baseurl}}/search-plugins/cross-cluster-search/#authentication-flow) in [Cross-cluster search]({{site.url}}{{site.baseurl}}/search-plugins/cross-cluster-search/). You must use a role that exists in both the remote and local clusters. The remote cluster must map the chosen role to the same username as in the local cluster.
```

File: _observing-your-data/ad/security.md (hunk 12 above)
Comment (vagimeli) [reply _observing-your-data/ad/security.md:74]: Revised

File: _observing-your-data/ad/index.md [hunk 15]
```diff
@@ -29,9 +29,14 @@ A detector is an individual anomaly detection task. You can define multiple dete
 1. Add in the detector details.
    - Enter a name and brief description. Make sure the name is unique and descriptive enough to help you to identify the purpose of the detector.
 1. Specify the data source.   
-   - For **Data source**, choose the index you want to use as the data source. You can optionally use index patterns to choose multiple indexes.
```
[the remark this answers is not in the document]
Comment (amitgalitz) [reply _observing-your-data/ad/index.md:32]: This singular usage sounds good to me on latest revision:
"- For **Data source**, choose one or more indexes to use as the data source. Alternatively, you can use an alias or index pattern to choose multiple indexes."

File: _observing-your-data/ad/security.md [hunk 16]
```diff
@@ -23,6 +23,11 @@ As an admin user, you can use the Security plugin to assign specific permissions
 
 The Security plugin has two built-in roles that cover most anomaly detection use cases: `anomaly_full_access` and `anomaly_read_access`. For descriptions of each, see [Predefined roles]({{site.url}}{{site.baseurl}}/security/access-control/users-roles#predefined-roles).
 
+If you use OpenSearch Dashboards to create your anomaly detectors, you may experience permission issues even with `anomaly_full_access`. This issue has been resolved in OpenSearch 2.17, but for earlier versions, the following additional permissions need to be added:
```
Comment (natebower) [inline _observing-your-data/ad/security.md:26]: "permissions issues"? Or do we actually mean "access issues"?

File: _observing-your-data/ad/security.md [hunk 17]
```diff
@@ -31,6 +36,42 @@ When a trigger generates an alert, the detector and monitor configurations, the
 
 To reduce the chances of unintended users viewing metadata that could describe an index, we recommend that administrators enable role-based access control and keep these kinds of design elements in mind when assigning permissions to the intended group of users. See [Limit access by backend role](#advanced-limit-access-by-backend-role) for details.
 
+### Selecting remote indexes with fine-grained access control
+
+To use a remote index as a data source for a detector, see the setup steps in [Authentication flow]({{site.url}}{{site.baseurl}}/search-plugins/cross-cluster-search/#authentication-flow) in [Cross-cluster search]({{site.url}}{{site.baseurl}}/search-plugins/cross-cluster-search/). You must use a role that exists in both the remote and local clusters. The remote cluster must map the chosen role to the same username as in the local cluster.
+
+---
+
+#### Example: Create a new user on local cluster
```
Comment (natebower) [inline _observing-your-data/ad/security.md:45]:
```suggestion
#### Example: Create a new user on the local cluster
```

File: _observing-your-data/ad/security.md [hunk 18]
```diff
@@ -31,6 +36,42 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
+
+1. Create a new user on the local cluster to use for creating detectors:
```
Comment (natebower) [inline _observing-your-data/ad/security.md:47]:
```suggestion
1. Create a new user on the local cluster to use for detector creation:
```

File: _observing-your-data/ad/security.md [hunk 19]
```diff
@@ -31,6 +36,42 @@ When a trigger generates an alert, the detector and monitor configurations, the
...
+curl -XPUT -k -u 'admin:<custom-admin-password>' 'https://localhost:9250/_plugins/_security/api/internalusers/anomalyuser' -H 'Content-Type: application/json' -d '{"password":"password"}'
+curl -XPUT -k -u 'admin:<custom-admin-password>' -H 'Content-Type: application/json' 'https://localhost:9250/_plugins/_security/api/rolesmapping/anomaly_full_access' -d '{"users" : ["anomalyuser"]}'
+```
+{% include copy-curl.html %}
+
+---
+
+### Custom results index
+
+To use a custom results index, you need additional permissions not included in the default roles provided by the OpenSearch Security plugin. To add these permission, see [Step 1: Define a detector]({{site.url}}{{site.baseurl}}/observing-your-data/ad/index/#step-1-define-a-detector) of the [Anomaly detection]({{site.url}}{{site.baseurl}}/observing-your-data/ad/index/) documentation.
```
Comment (natebower) [inline _observing-your-data/ad/security.md:73]:
```suggestion
To use a custom results index, you need additional permissions not included in the default roles provided by the OpenSearch Security plugin. To add these permissions, see [Step 1: Define a detector]({{site.url}}{{site.baseurl}}/observing-your-data/ad/index/#step-1-define-a-detector) in the [Anomaly detection]({{site.url}}{{site.baseurl}}/observing-your-data/ad/index/) documentation.
```

**Review by natebower — APPROVED** [review]
@amitgalitz @vagimeli Approved with one comment and a couple of changes. Thanks!

File: _observing-your-data/ad/security.md (hunk 16 above)
Comment (vagimeli) [inline _observing-your-data/ad/security.md:26]:
```suggestion
If you use OpenSearch Dashboards to create your anomaly detectors, you may experience access issues even with `anomaly_full_access`. This issue has been resolved in OpenSearch 2.17, but for earlier versions, the following additional permissions need to be added: 
```

**Pull request merged**